File: java/src/ik/OrderClassification.java
package ik;

/**
 * It was proven by Kohara and Suzuki that a graph with six or
 * fewer vertices is not intrinsically knotted.  This classification test
 * encodes that logic.
 */
public class OrderClassification implements IKClassification {
  public String classify(Graph graph) {    
    return (graph.getOrder() < 7) ? IS_NOT_IK : CANNOT_DETERMINE_IK;
  }
     
  public String getName() {
    return "OrderClassification";  
  }
  
  public String getDescription() {
    return "If there are 6 or less vertices then the graph is " +
           "NOT intrinsically knotted.";
  }
}

File: ruby/lib/ik_classifications/order_classification.rb
# It was proven by Kohara and Suzuki that a graph with six or
# fewer vertices is not intrinsically knotted.  This classification test
# encodes that logic.
class OrderClassification < IKClassification::Base
  def classify(graph)
    graph.order <= 6 ? IS_NOT_IK : CANNOT_DETERMINE_IK
  end
  
  def description
    "If there are 6 or less vertices then the graph is " +
    "NOT intrinsically knotted."
  end
end